holoviews/operation/datashader.py

```diff
@@ -400,7 +400,7 @@ def _process(self, element, key=None):
                 arrays.append(element.dimension_values(vd, flat=False))
             if len(arrays) > 1:
                 array = np.dstack(arrays).transpose([2, 0, 1])
-                coords['layer'] = range(len(arrays))
+                coord_dict['layer'] = range(len(arrays))
                 dims = ['layer'] + dims
```

Comment:
Could we use a more obscure name that conveys that users should never actually see 'layer'? This would also avoid clobbering a potentially user defined name...